### ISSUE TITLE
EIP Limits exceeded for AWS account

### ISSUE THREAD

**Issue by richardcase** [kubernetes/k8s.io#5390]
We are running into issues with some of the CAPA jobs running on the EKS prow cluster (and new AWS accounts handed out from boskos).

The account is **757254795865** and this is the error we are getting:

```
E0608 03:58:56.277164       1 awscluster_controller.go:282] controller/awscluster "msg"="failed to reconcile network" "error"="failed to create one or more IP addresses for NAT gateways: failed to allocate Elastic IP: AddressLimitExceeded: The maximum number of addresses has been reached.\n\tstatus code: 400, request id: 0160ac26-e9eb-4d3a-9dbe-414136a35964" "cluster"="k8s-upgrade-and-conformance-rpzcgh" "name"="k8s-upgrade-and-conformance-rpzcgh" "namespace"="k8s-upgrade-and-conformance-2r537n" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AWSCluster"
```

Could we request that the EIP limits are increased for this account (and any of the other AWS accounts handed out by boskos)?

/cc @Skarlso

**Comment by sbdtu5498** [comment]
I think it can be solved by increasing the ec2_max_elastic_ips [value](https://github.com/kubernetes/k8s.io/blob/main/infra/aws/terraform/service-quotas/modules/capa/variables.tf#LL55C6-L55C6). @richardcase @ameukam any suggestions on how much should we set the maximum value to. @hh any more suggestions on this one. It's my first time contributing in a infra based repo, so I am not very much aware of the workflows and can testing be done locally or not, etc. Thanks!

**Comment by pkprzekwas** [comment]
Good catch @sbdtu5498 . Unfortunately, as for today we don't have any workflows allowing contributors to interact with our AWS infrastructure... You can create a PR updating `ec2_max_elastic_ips` value to 150 or 200 and I'll try to apply changes. I'm not sure how eager AWS is with increasing this param... I expect them to send us query asking for explanation what we use those addresses for.

**Comment by sbdtu5498** [comment]
Sure @pkprzekwas, thanks a lot!

**Comment by dims** [comment]
@pkprzekwas when the support ticket gets created or pushed back, please DM it to me on slack. I'll see what i can do.

**Comment by pkprzekwas** [comment]
Sure, I still need to figure out creds to boskos-001 account... But will reach out to you as soon as possible @dims .

One thing that bothers me is the test @richardcase . Is it possible that it allocates over 100 IPs? Can we proof that? Should it allocate so many IPv4s? 

Maybe we have some problem with cleaning up the account after previous tests

**Comment by xmudrii** [comment]
The account in question is boskos-003 account. I just checked this account, there are 5 EIPs in that account (the account is currently being used by some test). Trying to create a new EIP results in the following error:

<img width="417" alt="image" src="https://github.com/kubernetes/k8s.io/assets/18719127/8aa3577b-4bca-4552-8bd0-629a13a67052">

This indicated that the actual EIP limit is only 5, even though AWS Support said we it's increased to 100. I'm going to follow up on the original support ticket and ask them if they can revisit this.

**Comment by xmudrii** [comment]
The limit for this account got temporarily increased to 20. I'll work with AWS support to get the limit increased to 100 as requested. Let's see if this is going to help at least a little bit to stabilize tests.

**Comment by xmudrii** [comment]
The limit is increased to 100. I also confirmed it via the Service Quotas dashboard.

<img width="923" alt="image" src="https://github.com/kubernetes/k8s.io/assets/18719127/88c61468-86d5-4a87-aac3-13c9bc82896d">

I'm going to close this issue since the issue should be solved now. If the problem persists, please feel free to reopen this issue.
